docs/ingressroute.md

```diff
@@ -0,0 +1,613 @@
+spec: 
+  virtualhost:
+    fqdn: root.bar.com
+  routes:
+    - match: /
+      services:
+        - name: s1
+          port: 80
+    # delegate the path, `/service2` to the IngressRoute object in this namespace with the name `service2`
+    - match: /service2 
+      delegate:
+        name: service2
+---
+# service2.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: service2
+  namespace: default
+spec: 
+  routes:
+    - match: /service2
+      services:
+        - name: s2
+          port: 80
+    - match: /service2/blog
+      services:
+        - name: blog
+          port: 80
+```
+
+### Across namespaces
+
+Delegation can also happen across Namespaces by specifying a `namespace` property for the delgate.  This is a particularly powerful paradigm for enabling multi-team Ingress management.
+
+In this example, the root IngressRoute has delegated configuration of paths matching `/blog` to the `blog` IngressRoute object in the `marketing` namespace.
+
+```yaml
+# root.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: namespace-delegation-root
+  namespace: default
+spec: 
+  virtualhost:
+    fqdn: ns-root.bar.com
+  routes: 
+    - match: /
+      services: 
+        - name: s1
+          port: 80
+# delegate the subpath, `/blog` to the IngressRoute object in the marketing namespace with the name `blog`
+    - match: /blog 
+      delegate:
+        name: blog
+        namespace: marketing
+---
+# blog.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: blog
+  namespace: marketing
+spec: 
+  routes: 
+    - match: /blog
+      services: 
+        - name: s2
+          port: 80
+```
+
+### Orphaned IngressRoutes
+
+It is possible for IngressRoute objects to exist that have not been delegated to by another IngressRoute.  These objects are considered "orphaned" and will be ignored by Contour in determining ingress configuration.
+
+### Restricted root namespaces
+
+IngressRoute delegation allows for Administrators to limit which users/namespaces may configure routes for a given domain, but it does not restrict where root IngressRoutes may be created. Contour has an enforcing mode which accepts a list of namespaces where root IngressRoutes are valid. Only users permitted to operate in those namespaces can therefore create IngressRoutes with the `virtualhost` field.
+
+This restricted mode is enabled in Contour by specifying a command line flag, `--ingressroute-root-namespaces`, which will restrict Contour to only searching the defined namespaces for root IngressRoutes.  This CLI accepts a comma separated list of namespaces where IngressRoutes are valid (e.g. `--ingressroute-root-namespaces=default,kube-system,my-admin-namespace`).
+
+IngressRoutes with a defined `virtualhost` field that are not in one of the allowed root namespaces will be flagged as `invalid` and will be ignored by Contour.
```

Comment:
Suggestion: please add a note that this is only supported for IngressRoute crds, --ingressroute-root-namespaces does not affect the operation of v1beta1.Ingress objects.